src/extensions.rs

```diff
@@ -270,6 +271,78 @@ impl RatchetTreeExtension {
     }
 }
 
+#[repr(u32)]
+#[cfg_attr(rustfmt, rustfmt_skip)]
+pub enum DeviceCapabilityType {
+    AddCap          = 0b0001u32,
+    RemoveCap       = 0b0010u32,
+    NonRemovableCap = 0b0100u32,
+    SelfUpdateCap   = 0b1000u32,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct DeviceCapabilities(pub u32);
+
+impl DeviceCapabilities {
```

Comment:
Can you put the new extension in it's own module?
It'll also need documentation.